src/sentry/seer/fetch_issues/by_function_name.py

```diff
@@ -309,114 +219,60 @@ def _get_projects_and_filenames_from_source_file(
+    if not file_projects:
+        logger.warning(
+            "No projects found for file. Using all projects.",
+            extra={"file": filename, "run_id": run_id},
         )
-        if issues:
-            logger.info("Found issues", extra=logger_extra | {"num_issues": len(issues)})
-        else:
-            logger.warning("No issues found", extra=logger_extra)
-        filename_to_issues[filename] = issues
+        file_projects_list = repo_projects.projects
 
-    return filename_to_issues
+    issues = _get_issues_for_file(
+        file_projects_list,
+        list(sentry_filenames),
+        [function_name],
+        event_timestamp_start,
+        event_timestamp_end,
+        max_num_issues_per_file=max_num_issues_per_file,
+        run_id=run_id,
+    )
+    group_ids = [issue["group_id"] for issue in issues]
+    return list(Group.objects.filter(id__in=group_ids).order_by("-last_seen"))
 
 
-def get_issues_related_to_file_patches(
-    *,
+def fetch_issues(
     organization_id: int,
     provider: str,
     external_id: str,
-    pr_files: list[PrFile],
+    filename: str,
+    function_name: str,
     max_num_issues_per_file: int = MAX_NUM_ISSUES_PER_FILE_DEFAULT,
     run_id: int | None = None,
-) -> dict[str, list[dict[str, Any]]]:
+) -> utils.SeerResponse:
     """
-    Get issues related to each file by matching filename-functions parsed from file patches with
-    filename-functions in the issue's event's stacktrace.
-
-    Assumes `pr_files` are from one repository.
-
-    Each issue includes its latest serialized event.
+    Fetch issues containing an event w/ a stacktrace frame that matches the `filename` and `function_name`.
     """
-
-    pr_files = safe_for_fetching_issues(pr_files)
-    pullrequest_files = [
-        PullRequestFile(filename=file["filename"], patch=file["patch"]) for file in pr_files
-    ]
-
-    filename_to_function_names = {}
-    for file in pullrequest_files:
-        file_extension = file.filename.split(".")[-1]
-        if file_extension not in patch_parsers_more:
-            logger.warning("No language parser", extra={"file": file.filename, "run_id": run_id})
-            continue
-        language_parser = patch_parsers_more[file_extension]
-        function_names = language_parser.extract_functions_from_patch(file.patch)
-        filename_to_function_names[file.filename] = list(function_names)
-
-    return get_issues_related_to_function_names(
-        organization_id=organization_id,
-        provider=provider,
-        external_id=external_id,
-        filename_to_function_names=filename_to_function_names,
+    repo_projects = utils.get_repo_and_projects(
+        organization_id, provider, external_id, run_id=run_id
+    )
+    groups = _fetch_issues_from_repo_projects(
+        repo_projects,
+        filename,
+        function_name,
         max_num_issues_per_file=max_num_issues_per_file,
         run_id=run_id,
     )
+    return utils.bulk_serialize_for_seer(groups)
```

Comment:
one thing we're losing here is the actual event ID which contains the function-filename. this info is only useful if the events in the issue are overgrouped

if it seems to be a problem for the agent then we can address by including it as an optional `IssueDetails` field and having seer it when the agent chooses to use the fetch-event tool